app/js/lib/intent-parser/index.js

```diff
@@ -1,28 +1,109 @@
-import ReminderParser from './reminder-parser';
-import QueryParser from './query-parser';
+import NormaliseCleaner from './cleaners/normalise-cleaner';
+import ContractionsCleaner from './cleaners/contractions-cleaner';
+import SalutationsCleaner from './cleaners/salutations-cleaner';
+import PolitenessCleaner from './cleaners/politeness-cleaner';
+
+import TimeParser from './parsers/time-parser';
+import UsersParser from './parsers/users-parser';
+import ActionParser from './parsers/action-parser';
+
+import ReminderRefiner from './refiners/reminder-refiner';
+import QueryRefiner from './refiners/query-refiner';
 
 const p = Object.freeze({
   // Properties
-  parsers: Symbol('parsers'),
+  normaliseCleaner: Symbol('normaliseCleaner'),
+  contractionsCleaner: Symbol('contractionsCleaner'),
+  salutationsCleaner: Symbol('salutationsCleaner'),
+  politenessCleaner: Symbol('politenessCleaner'),
+
+  timeParser: Symbol('timeParser'),
+  usersParser: Symbol('usersParser'),
+  actionParser: Symbol('actionParser'),
+
+  reminderRefiner: Symbol('reminderRefiner'),
+  queryRefiner: Symbol('queryRefiner'),
 });
 
 export default class IntentParser {
   constructor() {
-    this[p.parsers] = [
-      new ReminderParser(),
-      new QueryParser(),
-    ];
+    this[p.normaliseCleaner] = new NormaliseCleaner();
```

Comment:
Maybe these could be an array, we can reduce over them then and adding a new one is as easy as inserting into the array?